### ISSUE TITLE
Compiling failed, it seems missing `mozjs-102`.

### ISSUE THREAD

**Issue by Krysztal112233** [CinnamonSlackBuilds/csb#55]
### Slackware Version

15.0

### Cinnamon Version

5.8.2

### GIT Branch

main

### Description

Meson build cannot found `mozjs-102` while compiling.

### Compile log

```
The Meson build system
Version: 0.59.4
Source dir: /tmp/csb/cjs-5.8.0
Build dir: /tmp/csb/cjs-5.8.0/build
Build type: native build
Project name: cjs
Project version: 5.8.0
C compiler for the host machine: ccache cc (gcc 11.2.0 "cc (GCC) 11.2.0")
C linker for the host machine: cc ld.bfd 2.37-slack15
C++ compiler for the host machine: ccache c++ (gcc 11.2.0 "c++ (GCC) 11.2.0")
C++ linker for the host machine: c++ ld.bfd 2.37-slack15
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -fno-strict-aliasing: YES 
Compiler for C++ supports arguments -Wno-variadic-macros: YES 
Compiler for C++ supports arguments -Wno-missing-field-initializers: YES 
Compiler for C++ supports arguments -Wno-dangling-pointer: NO 
Compiler for C supports arguments -Wno-typedef-redefinition: NO 
Compiler for C++ supports link arguments -Bsymbolic-functions: YES 
Compiler for C supports arguments -fno-semantic-interposition: YES 
Compiler for C++ supports arguments -fno-semantic-interposition: YES 
Found pkg-config: /usr/bin/pkg-config (0.29.2)
Run-time dependency glib-2.0 found: YES 2.70.3
Run-time dependency gthread-2.0 found: YES 2.70.3
Run-time dependency gobject-2.0 found: YES 2.70.3
Run-time dependency gio-2.0 found: YES 2.70.3
Run-time dependency libffi found: YES 3.3
Run-time dependency gobject-introspection-1.0 found: YES 1.70.0
Found CMake: /usr/bin/cmake (3.21.4)
Run-time dependency mozjs-102 found: NO (tried pkgconfig and cmake)

../meson.build:131:0: ERROR: Dependency "mozjs-102" not found, tried pkgconfig and cmake
```

**Comment by Krysztal112233** [comment]
At the same time, I noticed that my system already installed `mozjs-78`.

```
root@server /h/k/D/b/c/cjs (master) [124]# slackpkg file-search mozjs

Looking for mozjs in package list. Please wait... DONE

The list below shows the packages that contains "mozjs" file.

[ installed ] - mozjs78-78.15.0esr-x86_64-1
```

**Comment by willysr** [comment]
From README:

1.  Clone the Git repo or download a tarball/zip of the repo.
    Master branch is recommended if you are using **current** and it's always tracking
    the latest release of Cinnamon project. If you are using **stable** version of Slackware,
    please use a branch matching your Slackware version; eg. 15.0.

**Comment by Krysztal112233** [comment]
Thanks for your patient replay, I’ll check it later 😊

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: Willy Sudiarto ***@***.***>
发送时间: 2023年10月28日 19:30
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [CinnamonSlackBuilds/csb] Compiling failed, it seems missing `mozjs-102`. (Issue #55)


From README:

  1.  Clone the Git repo or download a tarball/zip of the repo.
Master branch is recommended if you are using current and it's always tracking
the latest release of Cinnamon project. If you are using stable version of Slackware,
please use a branch matching your Slackware version; eg. 15.0.

**Comment by Krysztal112233** [comment]
I switched my cloned repository to `15.0` and fix all dependencies csb need, but at the last build it also report an error blow:

```
--2023-10-28 23:52:59--  https://github.com/GNOME/cogl/archive/1.22.8/cogl-1.22.8.tar.gz
Resolving github.com (github.com)... 224.0.0.1
Connecting to github.com (github.com)|224.0.0.1|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-10-28 23:53:02 ERROR 404: Not Found.

tar: /home/krysztal/Desktop/builds/CinnamonSlackBuild/cogl/cogl-1.22.8.tar.gz: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
server# 
```

The project named `cogl` has already deleted(or renamed, but I don't find any project related to `cogl` in GNOME  organization).

Should we change download link to official GNOME repository not mirrored repository in github?

**Comment by willysr** [comment]
Fixed in 15.0 branch now :)
let me know if i miss anything else

**Comment by SpreadiesInSpace** [comment]
Hi, I'm on Slackware Current and I seem to be running into the same issue with mozjs-102 dependency not being found.

I made sure to use the master branch. What might I have missed?

**Comment by willysr** [comment]
mozjs has been removed from Slackware-current, so i guess i will have to re-add it again

**Comment by willysr** [comment]
should be fixed with bed771899ff56b2bf9a9a38e84dd615a8ea29e0f

**Comment by SpreadiesInSpace** [comment]
> should be fixed with [bed7718](https://github.com/CinnamonSlackBuilds/csb/commit/bed771899ff56b2bf9a9a38e84dd615a8ea29e0f)

Confirmed working after mozjs-115 patch
